src/iuse_actor.cpp

```diff
@@ -2929,3 +2930,141 @@ void heal_actor::info( const item &, std::vector<iteminfo> &dump ) const
 
     dump.emplace_back( "TOOL", _( "<bold>Moves to use</bold>:" ), "", move_cost );
 }
+
+
+
+void place_trap_actor::data::load( JsonObject obj )
+{
+    assign( obj, "trap", trap );
+    assign( obj, "done_message", done_message );
+    assign( obj, "practice", practice );
+    assign( obj, "moves", moves );
+}
+
+void place_trap_actor::load( JsonObject &obj )
+{
+    assign( obj, "allow_underwater", allow_underwater );
+    assign( obj, "allow_under_player", allow_under_player );
+    assign( obj, "needs_solid_neighbor", needs_solid_neighbor );
+    assign( obj, "needs_neighbor_terrain", needs_neighbor_terrain );
+    assign( obj, "bury_question", bury_question );
+    if( !bury_question.empty() ) {
+        buried_data.load( obj.get_object( "bury" ) );
+    }
+    unburied_data.load( obj );
+    assign( obj, "outer_layer_trap", outer_layer_trap );
+}
+
+iuse_actor *place_trap_actor::clone() const
+{
+    return new place_trap_actor( *this );
+}
+
+bool is_solid_neighbor( const tripoint &pos, const int offset_x, const int offset_y )
+{
+    const tripoint a = pos + tripoint( offset_x, offset_y, 0 );
+    const tripoint b = pos - tripoint( offset_x, offset_y, 0 );
```

Comment:
Adding `points` to `tripoints` is possible. It is a bit less explicit, but looks cleaner.